File: tags/emberplussdk-1-4-0/tinyember/TinyEmberPlusRouter/TinyEmberPlusRouter/model/ParameterBase.cpp
#include "ParameterBase.h"

namespace model
{
   ParameterBase::ParameterBase(int number, Element* parent, std::string const& identifier, NotificationSink* notificationSink)
      : Element(number, parent, identifier)
      , m_notificationSink(notificationSink)
      , m_isReadOnly(false)
   {}
}

File: tags/emberplussdk-1-4-0/EmberLib.net/EmberPlusRouter/Model/ParameterBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace EmberPlusRouter.Model
{
   abstract class ParameterBase : Element
   {
      public ParameterBase(int number, Element parent, string identifier, Dispatcher dispatcher, bool isWriteable)
      : base(number, parent, identifier)
      {
         Dispatcher = dispatcher;
         IsWriteable = isWriteable;
      }

      public Dispatcher Dispatcher { get; private set; }
      public bool IsWriteable { get; private set; }
   }
}
